extensions/big-data-cluster/package.json

```diff
@@ -23,6 +23,56 @@
     "Microsoft.mssql"
   ],
   "contributes": {
+    "configuration": {
+      "type": "object",
+      "title": "Kubernetes configuration",
+      "properties": {
+          "mssql-bdc": {
+              "type": "object",
+              "description": "Kubernetes configuration",
+              "properties": {
+                  "mssql-bdc.kubectl-path": {
+                      "type": "string",
+                      "description": "File path to a kubectl binary."
+                  },
+                  "mssql-bdc.kubectl-path.windows": {
+                      "type": "string",
+                      "description": "File path to a kubectl binary."
+                  },
+                  "mssql-bdc.kubectl-path.mac": {
+                      "type": "string",
+                      "description": "File path to a kubectl binary."
+                  },
+                  "mssql-bdc.kubectl-path.linux": {
+                      "type": "string",
+                      "description": "File path to a kubectl binary."
+                  },
+                  "mssql-bdc.kubeconfig": {
+                      "type": "string",
+                      "description": "File path to the kubeconfig file."
+                  },
+                  "mssql-bdc.knownKubeconfigs": {
```

Comment:
Are these patterns all based on the vscode-kubernetes extension?

Reply:
Yes they expose their settings via the configs rather than manipulating the env var directly.